supervisor_agent.py

```diff
@@ -0,0 +1,233 @@
+import functools
```

Comment:
Is this file supposed to be merged to main? 
If not, please delete it.
If yes, please move to `/examples` or `/hack` directory.